### ISSUE TITLE
Windows binary cannot be built

### ISSUE THREAD

**Issue by ionash** [docker/cli#3294]
**Description**
I tried to build a Windows binary. But an error occurs.

**Steps to reproduce the issue:**
1. Clone `https://github.com/docker/cli.git`. (tag: `v20.10.8`)
2. Execute `docker buildx bake --set binary.platform=windows/amd64`

**Describe the results you received:**
The following error occurs.

```bash
 > [build 1/1] RUN --mount=ro --mount=type=cache,target=/root/.cache     --mount=from=dockercore/golang-cross:xx-sdk-extras,target=/xx-sdk,src=/xx-sdk     --mount=type=tmpfs,target=cli/winresources     xx-go --wrap &&     TARGET=/out ./scripts/build/binary &&     xx-verify $([ "static" = "static" ] && echo "--static") /out/docker:
#17 0.466 llvm-rc: Error parsing file: expected '-', '~', integer or '(', got PRODUCTVERSION
------
Dockerfile:49
--------------------
  48 |     ARG VERSION
  49 | >>> RUN --mount=ro --mount=type=cache,target=/root/.cache \
  50 | >>>     --mount=from=dockercore/golang-cross:xx-sdk-extras,target=/xx-sdk,src=/xx-sdk \
  51 | >>>     --mount=type=tmpfs,target=cli/winresources \
  52 | >>>     xx-go --wrap && \
  53 | >>>     # export GOCACHE=$(go env GOCACHE)/$(xx-info)$([ -f /etc/alpine-release ] && echo "alpine") && \
  54 | >>>     TARGET=/out ./scripts/build/binary && \
  55 | >>>     xx-verify $([ "$GO_LINKMODE" = "static" ] && echo "--static") /out/docker
  56 |     
--------------------
error: failed to solve: process "/bin/sh -c xx-go --wrap &&     TARGET=/out ./scripts/build/binary &&     xx-verify $([ \"$GO_LINKMODE\" = \"static\" ] && echo \"--static\") /out/docker" did not complete successfully: exit code: 1
Error: Process completed with exit code 1.
```

**Describe the results you expected:**
`docker buildx bake --set binary.platform=windows/amd64` builds a Windows binary.

**Additional information you deem important (e.g. issue happens only occasionally):**
The following file might be the cause of this issue.
```
scripts/build/.variables
```

The above file includes the following line.

```bash
VERSION=${VERSION:-$(git describe --match 'v[0-9]*' --dirty='.m' --always --tags | sed 's/^v//' 2>/dev/null || echo "unknown-version" )}
```

Since `--dirty` is set, `VERSION` becomes `20.10.8.m`. As a result, `scripts/winresources/common.rc` cannot be parsed.
If I remove the option `--dirty`, I can build a binary successfully.

**Output of `docker version`:**

```
Client:
 Version:           20.10.8+azure
 API version:       1.41
 Go version:        go1.16.7
 Git commit:        3967b7d28e15a020e4ee344283128ead633b3e0c
 Built:             Thu Jul 29 13:55:47 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.8+azure
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.7
  Git commit:       75249d88bc107a122b503f6a50e89c994331867c
  Built:            Fri Jul 30 01:30:57 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.9+azure
  GitCommit:        e25210fe30a0a703442421b0f60afac609f950a3
 runc:
  Version:          1.0.1
  GitCommit:        4144b63817ebcc5b358fc2c8ef95f7cddd709aa7
 docker-init:
  Version:          0.19.0
  GitCommit:
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., 0.6.3+azure)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 16
 Server Version: 20.10.8+azure
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: false
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2 io.containerd.runtime.v1.linux
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: e25210fe30a0a703442421b0f60afac609f950a3
 runc version: 4144b63817ebcc5b358fc2c8ef95f7cddd709aa7
 init version: 
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.8.0-1040-azure
 Operating System: Ubuntu 20.04.3 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 6.791GiB
 Name: fv-az38-392
 ID: CVYK:46VJ:RY4U:2ZQ7:KUAC:W2XA:CZWT:X2ZS:2ILO:27KD:42HL:IT2X
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: githubactions
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
I tried to build a binary by GitHub Actions. The workflow is the follwing.

```
jobs:
  build-test:
    runs-on: ubuntu-20.04
    steps:
      - name: checkout docker/cli
        uses: actions/checkout@v2
        with:
          repository: docker/cli
          ref: v20.10.8
          path: cli
      - name: create builder
        run: sudo docker buildx create --use
      - name: build binary
        working-directory: ./cli
        run: sudo docker buildx bake --set binary.platform=windows/amd64
```

**Comment by thaJeztah** [comment]
I think @mat007 had the same issue on a Windows machine, and it was due to the source files having the wrong line endings (CRLF instead of LF), which was solved by updating git's configuration before cloning the repository

**Comment by ionash** [comment]
I added `git diff --name-only` into `scripts/build/.variables` to confirm the modified files. As a result, I confirmed the following files. It seems that these files were deleted during the build step.

```
#17 0.578 .github/CODEOWNERS
#17 0.578 .github/ISSUE_TEMPLATE.md
#17 0.578 .github/PULL_REQUEST_TEMPLATE.md
#17 0.578 .github/workflows/codeql-analysis.yml
#17 0.578 .gitignore
#17 0.578 appveyor.yml
#17 0.578 cli/winresources/res_windows.go
```

```
```

**Comment by mat007** [comment]
> I think @mat007 had the same issue on a Windows machine

It seems to be a different issue here, I had a different error message.

And checking out `v20.10.8` I can confirm that I see the same error.
It does build correctly on `master` though.

**Comment by mat007** [comment]
On master
```
$ git describe --match 'v[0-9]*' --dirty='.m' --always --tags
v20.10.2-258-gfe2008d04e.m
```
whereas on `v20.10.8`
```
$ git describe --match 'v[0-9]*' --dirty='.m' --always --tags
v20.10.8.m
```

**Comment by mat007** [comment]
We could replace `--dirty='.m'` with `--dirty='-m'` maybe?

**Comment by mat007** [comment]
Oh no, I see. The regex for computing `VERSION_QUAD` is slightly wrong. I’ll make a fix.

**Comment by thaJeztah** [comment]
> It seems to be a different issue here, I had a different error message.

🤦 I didn't read carefully, and shouldn't do these things from my phone. I now see it's because of the "version" bit (building from a state with uncommitted changes).

I guess the script was anticipating to be used for releases (and at least, from a "clean" state)

**Comment by mat007** [comment]
This has actually already been fixed on master in #3048

**Comment by ionash** [comment]
Thank you very much! Now I understand that the right way to fix this issue is picking up the fixed script from `master`. I confirmed that it works on my environment also.
